transformation_test/transformation_test.go

```diff
@@ -90,50 +88,42 @@ func TestTransformationTests(t *testing.T) {
 				fmt.Sprintf("--config=%s", filepath.Join(tempPath, flbMainConf)),
 				fmt.Sprintf("--parser=%s", filepath.Join(filepath.Join(tempPath, flbParserConf))))
 
-			var stdout, stderr io.ReadCloser
-			stdout, err = cmd.StdoutPipe()
-			if err != nil {
-				t.Fatal("stdout pipe failure", err)
-			}
-			stderr, err = cmd.StderrPipe()
-			if err != nil {
-				t.Fatal("stderr pipe failure", err)
-			}
-
-			if err := cmd.Start(); err != nil {
-				t.Fatal("Failed to start command:", err)
+			var stdout, stderr bytes.Buffer
+			cmd.Stdout = &stdout
+			cmd.Stderr = &stderr
+			if err := cmd.Run(); err != nil {
+				t.Fatal("Failed to run command:", err)
 			}
 
-			// stderr and stdout need to be read in parallel to prevent deadlock
-			var eg errgroup.Group
-			eg.Go(func() error {
-				// read stderr
-				slurp, _ := io.ReadAll(stderr)
-				t.Logf("stderr: %s\n", slurp)
-				return nil
-			})
-
-			// read and unmarshal output
-			var data []map[string]any
-			out, _ := io.ReadAll(stdout)
-			_ = eg.Wait()
+			var d map[string]any
+			var data []map[string]interface{}
 
-			err := yaml.Unmarshal(out, &data)
-			if err != nil {
+			out := stdout.Bytes()
+			if err := yaml.Unmarshal(out, &d); err != nil {
 				t.Log(string(out))
 				t.Fatal(err)
 			}
-			// transform timestamp of actual results
-			for i, d := range data {
-				if date, ok := d["date"].(float64); ok {
-					date := time.UnixMicro(int64(date * 1e6)).UTC()
-					if date.After(testStartTime) {
-						data[i]["date"] = "now"
-					} else {
-						data[i]["date"] = date.Format(time.RFC3339Nano)
+
+			var entries []any
+			// Only search for entries if stdout is not null
+			if val, ok := d["entries"].([]any); ok {
+				entries = val
```

Comment:
I think this will fail if fluent-bit decides to use multiple WriteLogsRequests, no? (Probably you will fail at the `yaml.Unmarshal`, but if you don't, you'll just keep the entries from the last request here.)